### PR TITLE
Hardening phase SonarQube 6 - Daniel 2017-10-02

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ChangelogAction.java [hunk 1]
```diff
@@ -133,4 +146,131 @@ private static void writeParameters(JsonWriter json, ChangelogLoader.Change chan
...
+    private String inheritance;
+    private RuleKey ruleKey;
+    private String ruleName;
+    private final Map<String, String> params = new HashMap<>();
+
+    private Change() {
+    }
+
+    @VisibleForTesting
+    Change(String key, String type, long at, @Nullable String severity, @Nullable String userLogin,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ChangelogAction.java]: MAJOR Constructor has 9 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionTest.java [hunk 2]
```diff
@@ -52,16 +52,21 @@
...
 import static org.sonar.test.JsonAssert.assertJson;
 import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_KEY;
 import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_LANGUAGE;
 import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_ORGANIZATION;
 import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_QUALITY_PROFILE;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_SINCE;
 
 public class ChangelogActionTest {
 
+  System2 system2 = mock(System2.class);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionTest.java]: private, please use TestSystem2 instead

Comment (bartfastiel) [reply server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/ChangelogActionTest.java]: Done.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [issue](https://github.com/SonarSource/sonarqube/blob/978abd45f73cf9fec02ca904c3b60b8014fe26eb/server/sonar-db-dao/src/main/java/org/sonar/core/issue): Add a 'package-info.java' file to document the 'issue' package [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**